meson.build

```diff
@@ -116,6 +116,12 @@ if with_kernel_dir == ''
     with_kernel_dir = join_paths(path_prefix, 'lib', 'kernel')
 endif
 
+# Defaults to /usr/lib/initrd.d
+with_initrd_dir = get_option('with-initrd-dir')
```

Comment:
why introduce a separate directory for this? at any given moment, there should be just one such global initrd. why just not assume a single name for it, such as `/usr/lib/kernel/global-initrd` (i.e. `<KERNEL_DIR>/global-initrd`). The distro then chooses how to maintain it (override it on updates, keep it as symlink to a more descriptive name etc).

Reply:
Regarding being using other dir I tried to avoid break with the legacy behavior.

Why /usr/lib/initrd.d, honestly I started working with /usr/lib/kernel but since its not working with kernel version I though in keep it in other folder.

Reply:
i don't think it makes much sense to introduce another directory at `/usr/lib` level for this specific purpose. if you really need a directory, make it `/usr/lib/kernel/initrd.d` (`<KERNEL_DIR>/initrd.d`).

however, i also think that there should be just one file which the implementation would be looking at (as opposed to [cumbersome naming convention](https://github.com/clearlinux/clr-boot-manager/pull/102#discussion_r171961764)) that is currently in place. currently, we can only use one initrd, so make it a pre-defined name, e.g. `/usr/lib/kernel/global-initrd`. then a distro can name and place its initrds however it wants and you no longer need to have any concept of "initrd directory" in the first place.

anyways, i just thought that the current mechanism of picking the initrd is sloppy. i'll defer this to @bryteise.

Reply:
Hrm I guess I was looking for a short cut and was hoping systemd-boot chaining initrd work would come along faster than it mattered otherwise but I think we probably just need to fix this ourselves.

What do folks think about having clr-boot-manager assemble the distro initrds that might be on the system into the final initrd that is installed in /boot (assuming there is no user initrd that is overriding)?

Reply:
> What do folks think about having clr-boot-manager assemble the distro initrds that might be on the system into the final initrd that is installed in /boot (assuming there is no user initrd that is overriding)?

I thought about that as well, and I think the task might be worth a separate tool. We might be needing it elsewhere too (in the build, for example). It has to have some intelligence to distinguish various configurations (e.g. initrd from initramfs, different init schemes etc.). It seems that, maybe, that code does not belong here. Unless we want to simply `cat` two cpio archives (assuming they both use systemd for init).

Reply:
Hrm I was thinking it would only handle one scheme and be fairly limited (as I don't want to remake dracut) but I guess I don't have anything against it becoming more fully featured and not dracut in another tool. Perhaps a version one that is essentially cat and we can go from there to see if we would eventually adopt other features into another project/library?

Reply:
> Perhaps a version one that is essentially cat and we can go from there to see if we would eventually adopt other features into another project/library?

Yes, this makes sense for us (in Clear Linux).

Reply:
then, the old implementation which use an initrd linked to the kernel version will be removed?

Reply:
I don't know about removed. At least supplemented. My thought would be if a distro is using the kernel version initrds today they will still work after the change is completed (but they could then be using multiple initrds in addition to the kernel version linked one).